### PR TITLE
Replace com.google.common.base.Function (partly)

### PR DESCRIPTION
**Pull request by pks-1981**
Replace com.google.common.base.Function (partly)

For https://github.com/wiremock/wiremock/issues/2111

### CODE REVIEW

**Comment by timtebeek** [comment]
Are you making these changes by hand @pks-1981 ? You might want to have a look at these [Guava recipes](https://docs.openrewrite.org/reference/recipes/java/migrate/guava/noguava). I'm in the process of [covering Optional there as well](https://github.com/openrewrite/rewrite-migrate-java/pull/202).

**Comment by pks-1981** [comment]
> Are you making these changes by hand @pks-1981 ? You might want to have a look at these [Guava recipes](https://docs.openrewrite.org/reference/recipes/java/migrate/guava/noguava). I'm in the process of [covering Optional there as well](https://github.com/openrewrite/rewrite-migrate-java/pull/202).

Yes, only with my hands) I'm have more fun in doing it myself...

**Comment by timtebeek** [comment]
> > Are you making these changes by hand @pks-1981 ? You might want to have a look at these [Guava recipes](https://docs.openrewrite.org/reference/recipes/java/migrate/guava/noguava). I'm in the process of [covering Optional there as well](https://github.com/openrewrite/rewrite-migrate-java/pull/202).
> 
> Yes, only with my hands) I'm have more fun in doing it myself...

That's fine as well; even more appreciative of contributions made that way; Let's hope it's merged quickly without conflicts then!

File: src/main/java/com/github/tomakehurst/wiremock/stubbing/Scenario.java [hunk 1]
```diff
@@ -72,27 +69,15 @@ public String getState() {
...
-        .toSet();
+    List<String> requiredStates =
+        stubMappings.stream()
+            .map(StubMapping::getRequiredScenarioState)
+            .collect(Collectors.toList());
+
+    requiredStates.addAll(
+        stubMappings.stream().map(StubMapping::getNewScenarioState).collect(Collectors.toList()));
+
+    return requiredStates.stream().filter(Objects::nonNull).collect(Collectors.toSet());
```
Comment (pks-1981) [inline src/main/java/com/github/tomakehurst/wiremock/stubbing/Scenario.java:80]: Unfortunately, small changes in the order of the final collection.

AdminApiTest -> void getScenarios()
 - assertThat(body, jsonPartEquals("scenarios[0].possibleStates", asList("2", "3", "Started")));
+ assertThat(body, jsonPartEquals("scenarios[0].possibleStates", asList("Started", "2", "3")));

**Comment by pks-1981** [comment]
@timtebeek @tomakehurst waiting for review)

**Comment by pks-1981** [comment]
@tomakehurst 
maybe I'll merge my two MR?
if you're not planning merge soon...

**Comment by tomakehurst** [comment]
@pks-1981 I'm on holiday this week, so please bear with me. Appreciate you opening these, and I plan to review them soon.

**Review by oleg-nenashev — APPROVED** [review]
All changes look good to me though I mostly rely on CI

**Comment by oleg-nenashev** [comment]
One test fails in a single configuration, likely to be a flaky one

```
ProxyAcceptanceTest > fixedDelaysAreAddedToProxiedResponses() FAILED
    java.lang.AssertionError: 
    Expected: a value equal to or greater than <300L>
         but: <299L> was less than <300L>
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at com.github.tomakehurst.wiremock.ProxyAcceptanceTest.fixedDelaysAreAddedToProxiedResponses(ProxyAcceptanceTest.java:558)
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.github.tomakehurst.wiremock.common.xml.Xml (file:/D:/a/wiremock/wiremock/build/classes/java/main/) to constructor com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl()
WARNING: Please consider reporting this to the maintainers of com.github.tomakehurst.wiremock.common.xml.Xml
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

```

**Comment by pks-1981** [comment]
> One test fails in a single configuration, likely to be a flaky one
> 
> ```
> ProxyAcceptanceTest > fixedDelaysAreAddedToProxiedResponses() FAILED
>     java.lang.AssertionError: 
>     Expected: a value equal to or greater than <300L>
>          but: <299L> was less than <300L>
>         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
>         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
>         at com.github.tomakehurst.wiremock.ProxyAcceptanceTest.fixedDelaysAreAddedToProxiedResponses(ProxyAcceptanceTest.java:558)
> WARNING: An illegal reflective access operation has occurred
> WARNING: Illegal reflective access by com.github.tomakehurst.wiremock.common.xml.Xml (file:/D:/a/wiremock/wiremock/build/classes/java/main/) to constructor com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl()
> WARNING: Please consider reporting this to the maintainers of com.github.tomakehurst.wiremock.common.xml.Xml
> WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
> WARNING: All illegal access operations will be denied in a future release
> ```

"blinking" test?
Checked on my Windows 11.
image

**Pull request merged**